lib/xgu/xgu.h

```diff
@@ -712,6 +779,99 @@ uint32_t* xgu_set_light_local_attenuation(uint32_t* p, unsigned int light_index,
     return push_floats(p, v.f, 3);
 }
 
+/* ==== Texture stuff ==== */
+
+inline
+uint32_t* xgu_set_texture_offset(uint32_t* p, uint32_t offset) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_OFFSET, offset);
+}
+
+inline
+uint32_t* xgu_set_texture_format(uint32_t* p, uint8_t context_dma, bool cubemap_enable,
+                                 XguBorderSrc border_src, uint8_t dimensionality,
+                                 XguTexFormatColor format, uint8_t mipmap_levels,
+                                 uint8_t u_size, uint8_t v_size, uint8_t p_size) {
```

Comment:
These sizes (u, v, p ([huh? not w.. weird..]) are actually shifts (`1 << x` aka 2<sup>x</sup>).

The names should reflect this or we should accept `unsigned int` and calculate shifts within (or avoid this function and provide it through a wrapper with XGUX).

As I have commented elsewhere, these sizes are also ignored depending on the `format` (they are only used for swizzled  (`SZ`) textures).

Reply:
They're sometimes ignored; should xgu provide a default?